Gemfile

```diff
@@ -7,16 +7,16 @@ git_source(:github) do |repo_name|
 end
 
 # Core gems
-gem 'rails', '~> 5.2'
-gem 'pg', '~> 0.18' # Use postgresql as the database for Active Record
-gem 'puma', '~> 3.7' # Use Puma as the app server
+gem 'rails', '~> 6.0.1'
+gem 'pg' # Use postgresql as the database for Active Record
+gem 'puma' # Use Puma as the app server
 gem 'sassc-rails' # Use SASS for stylesheets
 gem 'slim-rails' # Use Slim for views
-gem 'uglifier', '>= 1.3.0' # Use Uglifier as compressor for JavaScript assets
+gem 'uglifier' # Use Uglifier as compressor for JavaScript assets
 
 gem 'rails_serve_static_assets' # Allow the heroku app to serve static files
-gem 'turbolinks', '~> 5' # Makes navigating your web application faster.
-gem 'redis', '~> 4.0' # Use Redis for caching
+gem 'turbolinks' # Makes navigating your web application faster.
+gem 'redis' # Use Redis for caching
```

Comment:
@Ardnived I'm quite fond of the alphabetical rule for the `Gemfile`. It does imply completely abandoning the effort to try and groups gems together though. What are your thoughts on this?

My take is that grouping of gems eventually always end up in subjective and/or arbitrary territory while simply sorting alphabetically provides some amount of organization which is easy to understand and follow.

Reply:
Sorting alphabetically is useful when you are looking for a specific gem, but grouping is helpful to people who are unfamiliar with the system and unfamiliar with the gems themselves. Grouping is subjective, but it conveys the intention with which the gem was added, and its place within the system.

To make it simple though. Alphabetical is best for experienced developers, and grouping is better for newcomers. I think the second situation is higher priority.

I think we should do as the hound suggestions, and sort alphabetically within each group.

Reply:
Hounds reference to `group` is actual blocks (like for test / development environment). It doesn't read linebreaks or comments as separate groupings.

So if we want to keep the grouping we should turn the check off.

Reply:
Are you sure about that? Rubocop docs seem to say otherwise. Groups are separated by an empty newline (or we could set it to be separated by comments if preferred).
https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Bundler/OrderedGems